yoyodyne/models/modules/__init__.py

```diff
@@ -0,0 +1,102 @@
+import argparse
+
+from ... import util
+from .base import BaseEncoder
+from .linear import LinearEncoder
+from .lstm import LSTMAttentiveDecoder, LSTMDecoder, LSTMEncoder  # noqa: F401
+from .transformer import TransformerDecoder  # noqa F401
+from .transformer import FeatureInvariantTransformerEncoder, TransformerEncoder
+
+
+class EncoderMismatchError(Exception):
+    pass
+
+
+def get_encoder_cls(encoder_arch: str, model_arch: str = None) -> BaseEncoder:
+    """Encoder factory.
+
+    Looks up module class for given encoder_arch string. If not found, backs
+        off to find compatible encoder for given model architecture.
+
+    Args:
+        encoder_arch (str).
+        model_arch (bool, optional).
+
+    Raises:
+        NotImplementedError.
+
+    Returns:
+        BaseEncoder.
+    """
+    encoder_fac = {
+        "feature_invariant_transformer": FeatureInvariantTransformerEncoder,
+        "linear": LinearEncoder,
+        "lstm": LSTMEncoder,
+        "transformer": TransformerEncoder,
+    }
+    model_to_encoder_fac = {
+        "attentive_lstm": LSTMEncoder,
+        "lstm": LSTMEncoder,
+        "pointer_generator_lstm": LSTMEncoder,
+        "transducer": TransformerEncoder,
+        "transformer": TransformerEncoder,
+    }
+    try:
+        model_cls = encoder_fac[encoder_arch]
+        util.log_info(f"Model: {model_cls.__name__}")
+        return model_cls
+    except KeyError:
+        util.log_info(
+            f"""Architecture {encoder_arch} not found. Looking for encoder
```

Comment:
Recommend a slight variation on this: something like:

`Architecture {encoder_arch} not compatible with model {model_arch}`